mllib/src/main/scala/org/apache/spark/ml/param/shared/SharedParamsCodeGen.scala

```diff
@@ -46,7 +46,10 @@ private[shared] object SharedParamsCodeGen {
         Some("\"probability\"")),
       ParamDesc[Double]("threshold",
         "threshold in binary classification prediction, in range [0, 1]",
-        isValid = "ParamValidators.inRange(0, 1)"),
+        isValid = "ParamValidators.inRange(0, 1)", finalMethods = false),
+      ParamDesc[Array[Double]]("thresholds",
+        "thresholds in multi-class classification prediction, must be array with size of classes.",
```

Comment:
This needs more explanation.  How about:
```
Thresholds for multi-class classification.  Rather than predicting the maximum rawPrediction value index as the class label, this re-weights the raw prediction for each class: rawPrediction(class) / thresholds(class).
```